### PR TITLE
refactor: add lightweight styling resources for Button control

### PR DESCRIPTION
**Pull request by Arieldelossantos**
﻿GitHub Issue: #https://github.com/unoplatform/Uno.Themes/issues/986

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## Description

Added Lightweight styling resources for Button control

### CODE REVIEW

File: src/library/Uno.Material/Styles/Controls/v2/Button.xaml [hunk 1]
```diff
@@ -13,21 +13,254 @@
...
+			<StaticResource x:Key="MaterialElevatedButtonDisabledBackground"
+							ResourceKey="OnSurfaceDisabledLowBrush" />
+			<StaticResource x:Key="MaterialElevatedButtonDisabledIconForeground"
+							ResourceKey="OnSurfaceDisabledBrush" />
+			<StaticResource x:Key="MaterialElevatedButtonDisabledContentForeground"
+							ResourceKey="OnSurfaceDisabledBrush" />
+			<StaticResource x:Key="MaterialElevatedButtonFocusedBackground"
+							ResourceKey="PrimaryFocusedBrush" />
+			<StaticResource x:Key="MaterialElevatedButtonPointerFocusedBackground"
+							ResourceKey="PrimaryFocusedBrush" />
```
[the remark this answers is not in the document]
Comment (kazo0) [reply src/library/Uno.Material/Styles/Controls/v2/Button.xaml:63]: Same for the rest of the file, needs to follow the pattern 

Proposed naming convention:
{?:style-variant}{control-name}{member}{?:visual-state}

| Name Part | Description |
| --- | --- |
| `style-variant` | **(Optional)** Certain styles have multiple variants. Ex: For Button we have style variants such as: Outlined, Text, Filled | 
| `control-name` | Name of the control type (Button, TextBox, CheckBox, etc.) |
| `member` | The property that this style should be assigned to (Foreground, Background, BorderBrush, etc.) |
| `visual-state` | **(Optional)**  Specifies which `VisualState` that this resource will be applied to (PointerOver, Checked, Disabled, etc.)  |

[the remark this answers is not in the document]
Comment (lesterbotello) [reply src/library/Uno.Material/Styles/Controls/v2/Button.xaml]: Didn't we say we were going to make the Theme resource names theme-agnostic? Based on this. we should also drop the `Material` part, right?

[the remark this answers is not in the document]
Comment (kazo0) [reply src/library/Uno.Material/Styles/Controls/v2/Button.xaml]: Correct

[the remark this answers is not in the document]
Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/Button.xaml]: fixed

File: src/library/Uno.Material/Styles/Controls/v2/Button.xaml [hunk 2]
```diff
@@ -13,21 +13,254 @@
...
+			<StaticResource x:Key="ElevatedButtonBackgroundDisabled"
+							ResourceKey="OnSurfaceDisabledLowBrush" />
+			<StaticResource x:Key="ElevatedButtonIconForegroundDisabled"
+							ResourceKey="OnSurfaceDisabledBrush" />
+			<StaticResource x:Key="ElevatedButtonContentForegroundDisabled"
+							ResourceKey="OnSurfaceDisabledBrush" />
+			<StaticResource x:Key="ElevatedButtonBackgroundFocused"
+							ResourceKey="PrimaryFocusedBrush" />
+			<StaticResource x:Key="ElevatedButtonBackgroundPointerFocused"
+							ResourceKey="PrimaryFocusedBrush" />
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/Button.xaml:63]: We should group the resources for each style by property. So all the "ElevatedButtonBackground[VISUAL_STATE]" brushes should be grouped together before the ButtonIcon and ButtonContent brushes. Same goes for the other button style resources below

File: src/library/Uno.Material/Styles/Controls/v2/Button.xaml
[the remark this answers is not in the document]
Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/Button.xaml:1]: fixed

File: src/library/Uno.Material/Styles/Controls/v2/Button.xaml [hunk 3]
```diff
@@ -13,21 +13,254 @@
...
+							ResourceKey="SurfaceBrush" />
+			<StaticResource x:Key="ElevatedButtonBackgroundPointerOver"
+							ResourceKey="PrimaryHoverBrush" />
+			<StaticResource x:Key="ElevatedButtonBackgroundPressed"
+							ResourceKey="PrimaryPressedBrush" />
+			<StaticResource x:Key="ElevatedButtonBackgroundDisabled"
+							ResourceKey="OnSurfaceDisabledLowBrush" />
+			<StaticResource x:Key="ElevatedButtonIconForegroundDisabled"
+							ResourceKey="OnSurfaceDisabledBrush" />
+			<StaticResource x:Key="ElevatedButtonContentForegroundDisabled"
```
[the remark this answers is not in the document]
Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/Button.xaml]: fixed

File: src/library/Uno.Material/Styles/Controls/v2/Button.xaml (hunk 2 above)
Comment (Arieldelossantos) [reply src/library/Uno.Material/Styles/Controls/v2/Button.xaml]: fixed

File: src/library/Uno.Material/Styles/Controls/v2/Button.xaml [hunk 4]
```diff
@@ -13,21 +13,530 @@
...
+	<x:String x:Key="ButtonHorizontalContentAlignment">Center</x:String>
+	<x:String x:Key="ButtonIconHorizontalAlignment">Stretch</x:String>
+	<x:String x:Key="ButtonIconVerticalAlignment">Stretch</x:String>
+	<x:String x:Key="IconButtonEllipseVerticalAlignment">Stretch</x:String>
+	<x:String x:Key="IconButtonEllipseHorizontalAlignment">Stretch</x:String>
+	<x:Boolean x:Key="ElevatedButtonIsTintEnabled">True</x:Boolean>
+
+	<ResourceDictionary.ThemeDictionaries>
+		<ResourceDictionary x:Key="Default">
+			<!--MaterialElevatedButtonStyle-->
```
Comment (kazo0) [inline src/library/Uno.Material/Styles/Controls/v2/Button.xaml:45]: Looks good! Only thing is the formatting, if you run Xaml Styler it should fix this, unless it's overridden in your XamlStyler settings but the x:Key and the ResourceKey should be on the same line for the StaticResource.

And one last thing, I know its annoying but we should also group the StaticResources for each style by property. So 

```xml
<!--MaterialElevatedButtonStyle-->

ElevatedButtonForeground
ElevatedButtonForegroundPointerOver
ElevatedButtonForegroundDisabled
ElevatedButtonForegroundPressed

ElevatedButtonIconForeground
ElevatedButtonIconForegroundPointerOver
ElevatedButtonIconForegroundDisabled
ElevatedButtonIconForegroundPressed

etc...
```

Similar to what we do for other styles like [here](https://github.com/unoplatform/Uno.Themes/blob/d5d803370cc99e11e3e895992a220b95abec477a/src/library/Uno.Material/Styles/Controls/v2/TextBox.xaml#L27-L50)

**Review by kazo0 — APPROVED** [review]

**Pull request merged**